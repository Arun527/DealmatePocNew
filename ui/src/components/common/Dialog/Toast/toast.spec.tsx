import React from "react";
import TestContextProvider from "../../../../test/test-container";
import { ToastContextProvider, useToastContext } from "./toast";
import { render } from "@testing-library/react";
import { waitForElementByText } from "../../../../test/test-util";

const toastMessage = {
  severity: "success",
  detail: "tested successfully",
};

const TestComponent: React.FunctionComponent = () => {
  const showToast = useToastContext();

  return <button onClick={() => showToast(toastMessage)}>Show Toast</button>;
};

describe("Toast-Context", () => {
  let baseElement, screen;

  beforeEach(() => {
    ({ baseElement, ...screen } = render(
      <ToastContextProvider children={<TestComponent />} />,
      { wrapper: TestContextProvider }
    ));
  });

  it("Match snapshots", () => {
    expect(baseElement).toMatchSnapshot();
  });

  it("should show toast", async () => {
    const button = await waitForElementByText("Show Toast");
    button.click();
    const message = await waitForElementByText("tested successfully");
    expect(message?.textContent.trim()).toBe("tested successfully");
  });
});
