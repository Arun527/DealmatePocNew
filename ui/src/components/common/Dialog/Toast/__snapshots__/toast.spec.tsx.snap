// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast-Context Match snapshots 1`] = `
<body>
  <div>
    <div
      class="p-toast p-component p-toast-top-right p-ripple-disabled"
      data-pc-name="toast"
      data-pc-section="root"
      style="position: fixed; top: 20px; right: 20px;"
    >
      <div />
    </div>
    <button>
      Show Toast
    </button>
  </div>
</body>
`;
